sympy/tensor/tensor_fields.py

```diff
@@ -0,0 +1,797 @@
+# -*- coding: utf-8 -*-
+
+from sympy.matrices import Matrix, eye
+from sympy.core import diff, Add, Symbol
+from sympy.simplify import simplify
+from sympy.tensor.arraypy import Arraypy, Tensor, matrix2arraypy, \
+    matrix2tensor, list2arraypy, list2tensor
+from sympy.tensor.tensor_methods import is_asymmetric
+from sympy import sqrt
+
+"""Module tensor_fields contains functions for working with the tensor fields:
+-the calculation of the differential and the gradient of the function;
+-curl and divergence of a vector field;
+-the calculation of the derivative of Li;
+-the calculation the Lie brackets of two vector fields;
+-the calculation the external differentiation of differential forms.
+
+All functions can be operated with input arguments specified as a tensor.
+Some of the input arguments of functions can be a list, matrix or array of
+arraypy. The starting index of tensor may not be equal to 0. In such instance,
+the object, which returns the function will have the starting index
+corresponding to the conditions of the call.
+Also lie_w function and dw involves working with multidimensional arrays.
+
+Functions are work with the multidimensional arrays arraypy and tensors,
+classes and methods which are contained in the module arraypy.
+
+"""
+
+
+def df(f, args, output_type='l'):
+    """Return an the 1-form df, differential of function f(x).
+
+    Examples:
+    =========
+
+    >>> from sympy.tensor.tensor_fields import df
+    >>> from sympy import symbols, sin
+    >>> from sympy.tensor.arraypy import Arraypy
+    >>> x1, x2, x3= symbols('x1 x2 x3')
+
+    f it’s a function the differential of that is calculated:
+
+    >>> f=x1**2*x2 + sin(x2*x3 - x2)
+
+    args it’s a list of symbol arguments of the function f. It can be in list,
```

Comment:
what kind of character is this `’`. Vim gives me unicode representation _e2 80 99_, while ordinary `'` is _27_.